observability/src/main/java/io/grpc/observability/interceptors/LogHelper.java

```diff
@@ -300,53 +317,85 @@ void logCancel(
         .setEventLogger(eventLogger)
         .setLogLevel(LogLevel.LOG_LEVEL_DEBUG)
         .setRpcId(rpcId);
-    sink.write(logEntryBuilder.build());
+    sink.write(logEntryBuilder.build(), locationTags, customTags);
   }
 
   GrpcLogRecord.Builder createTimestamp() {
     long nanos = timeProvider.currentTimeNanos();
     return GrpcLogRecord.newBuilder().setTimestamp(Timestamps.fromNanos(nanos));
   }
 
+  private static final Set<String> NEVER_INCLUDED_METADATA = new HashSet<>(
+      Collections.singletonList(
+          // grpc-status-details-bin is already logged in a field of the binlog proto
+          STATUS_DETAILS_KEY.name()));
+  private static final Set<String> ALWAYS_INCLUDED_METADATA = new HashSet<>(
+      Collections.singletonList(
+          "grpc-trace-bin"));
+
   static final class PayloadBuilder<T> {
     T payload;
     int size;
+    boolean truncated;
 
-    private PayloadBuilder(T payload, int size) {
+    private PayloadBuilder(T payload, int size, boolean truncated) {
       this.payload = payload;
       this.size = size;
+      this.truncated = truncated;
     }
   }
 
   // TODO(dnvindhya): Create a unit test for the metadata conversion
```

Comment:
Is the TODO still applicable? When will the test be added?

Reply:
No, tests are added. Removed TODO